kafka/producer_test.go

```diff
@@ -179,6 +179,41 @@ func TestProducerAPIs(t *testing.T) {
 	if offsets != nil {
 		t.Errorf("OffsetsForTimes() failed but returned non-nil Offsets: %s\n", offsets)
 	}
+
+	p.Close()
+	err = p.Produce(&Message{TopicPartition: TopicPartition{Topic: &topic2, Partition: 0},
+		Value: []byte("Producer Validation"), Key: []byte("This is my key")},
+		nil)
+	expectedErr := Error{code: ErrState, str: "not a valid Producer state."}
+	if err != expectedErr {
```

Comment:
The if-else clauses can be made into a function, will clean this up quite a bit.